cds_books/literature/tasks.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019-2020 CERN.
+#
+# CDS Books is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+"""Literature tasks."""
+
+import logging
+import urllib
+
+from celery import shared_task
+from invenio_app_ils.documents.api import Document
+from invenio_db import db
+
+from .covers import build_literature_cover_urls
+
+logger = logging.getLogger('tasks')
+
+# NOTE: Series have covers too
+# from invenio_app_ils.records.api import Series
+
+
+ALLOWED_CLASSES = [Document]
+MIN_CONTENT_LENGTH = 128
+
+
+def update_cover(sender, *args, **kwargs):
+    """Triggers async task to update cover metadata of the record."""
+    record = kwargs.get("record", {})
+    if record.__class__ in ALLOWED_CLASSES:
```

Comment:
if you have the schema field, it is better to check that, because you might override the class.